microcosm_sagemaker/testing/pickle_comparison.py

```diff
@@ -0,0 +1,95 @@
+from functools import partial
+from io import BytesIO
+from pathlib import Path
+
+import numpy as np
+from hamcrest.core.base_matcher import BaseMatcher
+from sklearn.externals import joblib
+
+from microcosm_sagemaker.testing.bytes_extractor import ExtractorMatcherPair
+
+
+class IsObjectEqual(BaseMatcher):
+    """
+    Matcher to compare the object blob saved at filepath A with another object blob saved
+    at filepath B.
+    Sometimes there is some randomness within these generated files, so the user can optionally
+    provide a `ignore_properties` array which will ignore mismatches of those given object
+    properties.  For instance, if we have:
+    MyObject(foo=bar, extras=Test)
+    MyObject(foo=bar, extras=Test2)
+    If ignore_properties is not specified, these will be unequal.  If ignore_properties is
+    set to ["extras"], they will be equal.
+    """
+    def __init__(self, obj, ignore_properties=None):
+        self.ignore_properties = ignore_properties
+        self.compare_object = obj
+
+    def _matches(self, item):
+        compare_a = self._get_dict_structure(self.compare_object)
+        compare_b = self._get_dict_structure(item)
+
+        total_attributes = (
+            attr
+            for attr in set(compare_a.keys()) | set(compare_b.keys())
+            if attr not in self.ignore_properties
+        )
+
+        for attr in total_attributes:
+
+            if attr not in compare_a or attr not in compare_b:
+                return False
+
+            if type(compare_a[attr]) != type(compare_b[attr]):  # noqa: 721
+                return False
+
+            if isinstance(compare_a[attr], np.ndarray):
+                if (compare_a[attr] != compare_b[attr]).any():
+                    return False
+            elif compare_a[attr] != compare_b[attr]:
+                return False
+
+        return True
+
+    def _get_dict_structure(self, obj):
+        if isinstance(obj, dict):
+            return obj
+        return vars(obj)
+
+
+def pickle_extractor(binary):
+    return joblib.load(BytesIO(binary))
+
+
+def _matches_with_object(obj, ignore_properties=None):
```

Comment:
```suggestion
def matches_object(
    obj: Union[object, dict],
    ignore_properties: Optional[List[str]]=None,
):
```

I think this could be generally useful